spec/integration/matcher_examples_spec.rb

```diff
@@ -687,10 +689,8 @@
           band.records.where(producers: 100..200).first
         end
 
-        it 'raises InvalidQuery' do
-          lambda do
-            found_record
-          end.should raise_error(Mongoid::Errors::InvalidQuery)
+        it 'finds' do
+          expect(found_record).to eq(record)
```

Comment:
Now, this definitely changes the outcome of the test, but this looks ultimately the same as the test on 664 and it feels like it should return the correct record. This seems like an improvement, but I don't know a lot about the matcher code so @p-mongo let me know if this is incorrect.

Note: this does not fix MONGOID-4911.

Reply:
This test uses simple values that aren't subject to MONGOID-4911 and therefore it does pass. The change you've made in this PR is to apply existing Range expansion to embedded matching, which I think is fine if we are going to then work on MONGOID-4911 and repair how it works.

I would leave the current test as you have it but add a test for `band.records.where(producers: 199..200).first` which would find the band currently but should return nil, and mark it pending on MONGOID-4911.

Reply:
I added this test in MONGOID-4911 PR.